papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,553 @@
+Bundling is what allows the script to contain multiple jobs executing the same operation.
+By storing information on individual bundles, signac-flow prevents accidental resubmission just as in the unbundled case.
+While the example above does not use either groups or aggregation, bundles works seamlessly with both.
+
+.. figure:: signac-flow_aggregation-groups-bundling.pdf
+    :align: center
+    :scale: 90 %
+    :figclass: w
+
+    Aggregation, groups, and bundling allow users to build complex workflows.
+    The features are orthogonal, and can be used in any combination.
+    Aggregation enables one operation or group to act on multiple jobs.
+    Groups allow users to combine multiple operations into one, with dependencies among operations resolved at run time.
+    Bundling helps users efficiently leverage HPC schedulers by submitting multiple commands in the same script, to be executed in serial or parallel.
+
+Synced Collections: Backend-agnostic, persistent, mutable data structures
+-------------------------------------------------------------------------
+
+Motivation
+~~~~~~~~~~
+
+All of signac's principal functions are designed around efficiently indexing a collection of directories.
+By organizing job directories by the hash of their state point, signac can perform many operations in constant time.
+To present a Pythonic API, state points are exposed via a dictionary-like interface, making it very easy to modify a state point and have that change transparently reflected in both the JSON file and the name of the corresponding directory.
+
+The need to parse these JSON files for indexing and the complexity of modifying them represent the most significant barriers to scaling signac.
+Even in the absence of file modification, reading a large number of files simply to produce a database index becomes prohibitively expensive for large data spaces.
+Although various optimizations have incrementally improved signac's scalability, an alternative means of storing the state point and associated metadata that circumvents the heavy I/O costs of our current approach has the potential to make a much larger impact.
+However, replacing individual JSON files as the primary data source for signac without breaking signac's API would require a generic means for providing the same interface to the underlying index and metadata irrespective of the underlying storage mechanism.
+Once developed, however, such an API would abstract out enough of the internals of signac to enable other generalizations as well, such as making it relatively easy to support alternate (and nearly arbitrary) data space layouts.
+
+The synced collections subpackage of signac represents the culmination of our efforts to expose this functionality, providing a generic framework within which interfaces corresponding to any of Python's built-in types can be easily constructed with arbitrary underlying synchronization protocols.
```

Comment:
Is it too strong to say _any_ of Python's built-in types? Should it be:
```suggestion
The synced collections subpackage of signac represents the culmination of our efforts to expose this functionality, providing a generic framework within which interfaces corresponding to Python's ``list`` and ``dict`` data structures can be easily constructed with arbitrary underlying synchronization protocols.
```

Reply:
`set` needs to be added as well.

Reply:
We don't have any synced collections supporting `set` to my knowledge.

Reply:
We cannot currently build sets, no. I intended the language to indicate more what would be possible than what has already been built, otherwise arbitrary underlying synchronization protocols also doesn't make sense. This section is talking more about what the framework aspires to be, and what is possible within the structure. The details of what is currently implemented are IMO not terribly important since adding new stuff is actually quite easy now that the framework is set up.

Reply:
This has been edited and is outdated. Resolving.